wordpress-mobile/WordPress-iOS
# My Site Dashboard: handle local and remote cards

Part of #17873

**Important: handling errors, persistence, update, pull to refresh, etc are not part of this task.**

<img src="https://user-images.githubusercontent.com/7040243/153008316-3f20d1e2-587d-4c58-9953-dbde674bb886.png" width="300">

This PR handles:

* Local cards
* Remote cards
* Configuring them

I will probably expand on the architecture in a future post, but the idea is quite simple: cards are registered in `DashboardCard` and they can be: 1) remote or 2) local. Remote cards are powered by remote data, while local cards aren't.

### To test

Make sure to run your sandbox with the 1.1 API diff applied and to have the My Site Dashboard flag enabled:

1. Open the app
2. Tap "Dashboard"
3. Notice that the Quick Actions row appear
4. Posts card might appear if 1) you have draft posts or 2) you have scheduled posts

For (4) you can play by creating drafts/scheduled posts and closing and reopening the app (updates are not handled yet).

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added unit tests for the parser.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the parser.

PR submission checklist:
